jetty-webapp/src/test/java/org/eclipse/jetty/webapp/HugeResourceTest.java

```diff
@@ -244,6 +244,26 @@ public void testDownload(String filename, long expectedSize) throws Exception
         }
     }
 
+    @ParameterizedTest
+    @MethodSource("staticFiles")
+    public void testHead(String filename, long expectedSize) throws Exception
+    {
+        URI destUri = server.getURI().resolve("/" + filename);
+        InputStreamResponseListener responseListener = new InputStreamResponseListener();
+
+        Request request = client.newRequest(destUri)
+            .method(HttpMethod.HEAD);
+        request.send(responseListener);
+        Response response = responseListener.get(5, TimeUnit.SECONDS);
```

Comment:
`responseListener.get()` only waits for the headers, so to be sure here I would also do (maybe in a try-with-resources):

`assertEquals(-1, response.getInputStream().read());`

which would guarantee that the HEAD response really came without body, as it should be.

Reply:
Good point.  
But if that fails, does that mean that both jetty server **and** jetty client have bugs then? ;-)

Reply:
Does not matter - we will have a failed test and we'll investigate :)

Reply:
@joakime I'll fix this to get this merged ASAP.